File: box/src/main/java/de/qabel/box/storage/StorageReadBackend.java
package de.qabel.box.storage;

import de.qabel.box.storage.exceptions.QblStorageException;

public interface StorageReadBackend extends AuthenticatedDownloader {

    /**
     * Download a file from the storage
     */
    StorageDownload download(String name) throws QblStorageException;

    /**
     * Download a file from the storage if it was modified (new version / etag / ...)
     */
    @Override
    StorageDownload download(String name, String ifModifiedVersion) throws QblStorageException, UnmodifiedException;

    String getUrl(String meta);
}

File: box/src/main/java/de/qabel/box/storage/StorageReadBackend.kt
package de.qabel.box.storage

import de.qabel.box.storage.exceptions.QblStorageException

interface StorageReadBackend : AuthenticatedDownloader {

    /**
     * Download a file from the storage
     */
    @Throws(QblStorageException::class)
    fun download(name: String): StorageDownload

    /**
     * Download a file from the storage if it was modified (new version / etag / ...)
     */
    @Throws(QblStorageException::class, UnmodifiedException::class)
    override fun download(name: String, ifModifiedVersion: String): StorageDownload

    fun getUrl(meta: String): String
}
